60.md

```diff
@@ -0,0 +1,188 @@
+```jsonc
+{
+    "kind": 37375,
+    "content": nip44_encrypt([
+        [ "balance", "100", "sat" ],
+        [ "privkey", "hexkey" ] // explained in NIP-61
+    ]),
+    "tags": [
+        [ "d", "my-wallet" ],
+        [ "mint", "https://mint1" ],
+        [ "mint", "https://mint2" ],
+        [ "mint", "https://mint3" ],
+        [ "name", "my shitposting wallet" ],
+        [ "unit", "sat" ],
+        [ "description", "a wallet for my day-to-day shitposting" ],
+        [ "relay", "wss://relay1" ],
+        [ "relay", "wss://relay2" ],
+    ]
+}
+```
+
+The wallet event is a parameterized replaceable event `kind:37375`.
+
+Tags:
+* `d` - wallet ID.
+* `mint` - Mint(s) this wallet uses -- there MUST be one or more mint tags.
+* `relay` - Relays where the wallet and related events can be found. -- one ore more relays SHOULD be specified. If missing, clients should follow [[NIP-65]].
+* `unit` - Base unit of the wallet (e.g. "sats", "usd", etc).
+* `name` - Optional human-readable name for the wallet.
+* `description` - Optional human-readable description of the wallet.
+* `balance` - Optional best-effort balance of the wallet that can serve as a placeholder while an accurate balance is computed from fetching all unspent proofs.
+* `privkey` - Private key used to unlock P2PK ecash. MUST be stored encrypted in the `.content` field. **This is a different private key exclusively used for the wallet, not associated in any way to the user's nostr private key** -- This is only used when receiving funds from others, described in NIP-61.
+
+Any tag, other than the `d` tag, can be [[NIP-44]] encrypted into the `.content` field.
+
+### Deleting a wallet event
+Due to PRE being hard to delete, if a user wants to delete a wallet, they should empty the event and keep just the `d` identifier and add a `deleted` tag.
+
+## Token Event
+Token events are used to record the unspent proofs that come from the mint.
+
+There can be multiple `kind:7375` events for the same mint, and multiple proofs inside each `kind:7375` event.
+
+```jsonc
+{
+    "kind": 7375,
+    "content": nip44_encrypt({
+        "mint": "https://stablenut.umint.cash",
+        "proofs": [
+            {
+                "id": "005c2502034d4f12",
+                "amount": 1,
+                "secret": "z+zyxAVLRqN9lEjxuNPSyRJzEstbl69Jc1vtimvtkPg=",
+                "C": "0241d98a8197ef238a192d47edf191a9de78b657308937b4f7dd0aa53beae72c46"
+            }
+        ]
+    }),
+    "tags": [
+        [ "a", "37375:<pubkey>:my-wallet" ]
+    ]
+}
+```
+
+`.content` is a [[NIP-44]] encrypted payload storing the mint and the unencoded proofs.
+* `a` an optional tag linking the token to a specific wallet.
+
+### Spending proofs
+When one or more proofs of a token are spent, the token event should be [[NIP-09]]-deleted and, if some proofs are unspent from the same token event, a new token event should be created rolling over the unspent proofs and adding any change outputs to the new token event.
```

Comment:
This is challenging to maintain on the client since each spend requires the client to determine which events need to be deleted and what proofs from the deleted event should roll over to the new event while possibly adding more proofs.

What if the `Spending History Event` had inputs of the spent proofs and outputs of the new proofs instead? This event would now be mandatory, but it would make wallet tracking easier on the client since all the client has to do is replay all the events to get the current state. When the wallet makes a transaction, it is an easy-to-construct single event.

Maybe optionally, there can be a wallet "snapshot" or backup event that has all the unspent proofs at a specified point in time. The wallet can then replay the spending events since then to get the current state.

Reply:
> This is challenging to maintain on the client since each spend requires the client to determine which events need to be deleted and what proofs from the deleted event should roll over to the new event while possibly adding more proofs.
> 
This is not problematic at all, or maybe I'm not understanding your point; but if the client fetches, say, 4 events, which each has, say, 3 proofs inside, total of 12 proofs.

If the client spends, say, 10 of those proofs, it's trivial to know it has to rollover the 2 unspent proofs into a new token event and delete all 4 token events.

Which equates to the client publishing 2 events:
Kind 9 (event deletion) e-tagging 4 events
Kind 7375 storing two proofs.

Just to be very explicit, the relays won't send the 4 deleted events any more when a client REQs for 7375s, it would now just reply with a single event, the one that contains the two proofs.

Again, perhaps I misunderstood your comment 😂

Reply:
I tend to agree with @davidcaseria 
The annoying part with deleting events is if you have two nip60 wallet implementations on your device or different device.
You would store those events/proofs in an indexdb or local storage. 
To get these 2 wallets in sync means replaying all the unspent proof events from the wallet creation (to account for the deleted one that once existed) Which you probably don’t want to do each time you open your app.

the append only method sounds simpler and more robust to sync the wallet accross platform/app, you just take all the unspent proofs events minus the spent proofs event and you should be fine.

it doesn’t change anything for that NIP except that the spent proofs feed/event would now be mandatory. One can still delete the unspent proofs event but this would now be optional.

Reply:
Also if the wallet wants to show an history for the user, which most neobanks do (wise, Revolut etc) keeping the full event history is a must

Reply:
I have the feeling the NIP is getting too deep into implementation details at this point, which isn't the goal of a NIP. It should be an implementation decision how to track the balance of the wallet. I'd find great to make the `Spending History Event` mandatory and thus wallets could choose an event sourcing approach as proposed by @davidcaseria too.